src/colorfulness.jl

```diff
@@ -0,0 +1,78 @@
+"""
+
+    HASLER_AND_SUSSTRUNK_M3 <: NoReferenceIQI
+    M =  hasler_and_susstrunk_m3(img)
+    M =  colorfulness(img)
+
+Calculates the colorfulness of a natural RGB image according to the
+metric, M3 from [1]. As a guide to interpretation of results, the
+authors suggest:
+
+|Not colorful        |  0|
+|slightly colorful   | 15|
+|moderately colorful | 33|
+|averagely colorful  | 45|
+|quite colorful      | 59|
+|highly colorful     | 82|
+|extremely colorful  |109|
+
+[1] Hasler, D. and Süsstrunk, S.E., 2003, June. Measuring colorfulness
+in natural images. In Human vision and electronic imaging VIII
+(Vol. 5007, pp. 87-96). International Society for Optics and
+Photonics.
+
+"""
+struct HASLER_AND_SUSSTRUNK_M3 <: NoReferenceIQI end
+
+# api
+(iqi::HASLER_AND_SUSSTRUNK_M3)(img) = hasler_and_susstrunk_m3(img)
+
+@doc (@doc HASLER_AND_SUSSTRUNK_M3)
+function hasler_and_susstrunk_m3(img::AbstractArray{<:AbstractRGB})
+
+    R = 255 .* float(red.(img))
+    G = 255 .* float(green.(img))
+    B = 255 .* float(blue.(img))
+
+    rg = R .- G 
+    μrg, σrg = mean(rg), std(rg)
+   
+    yb = 0.5 .* (R .+ G) .- B
+    μyb,  σyb = mean(yb), std(yb)
+
+    μrgyb = sqrt(μrg^2 + μyb^2)
+    σrgyb = sqrt(σrg^2 + σyb^2)
+
+    return σrgyb + 0.3 * μrgyb
+
+end
+
+@doc (@doc HASLER_AND_SUSSTRUNK_M3)
```

Comment:
This is redundant -- If you type `? hasler_and_susstrunk_m3` then there would be two identical docstrings showing up by default.

```julia
"""some docs"""
struct FOO end

@doc (@doc FOO)
foo(x::Number) = println("a number")

@doc (@doc FOO)
foo(x::String) = println("a string")
```

`?foo` prints:

```julia
  some docs

  ────────────────────────────────────────────────────────────────────────────────────────────────

  some docs
```

Reply:
Fixed.

Reply:
Ha, I might not express myself well.

The `@doc (@doc HASLER_AND_SUSSTRUNK_M3)` trick copies all the docstring of `HASLER_AND_SUSSTRUNK_M3` to two methods of `hasler_and_susstrunk_m3`. So that when users type `?hasler_and_susstrunk_m3` it would print all possible method docstrings of `hasler_and_susstrunk_m3`. In the previous example, it prints "some docs" *twice*, which is definitely not what we want.

I'll fix it later.